src/block.cpp

```diff
@@ -1,18 +1,20 @@
 #include "block.h"
 
 Block::Block()
+: id(0)
+, cells()
+, cellSize(30)
+, rotationState(0)
+, rowOffset(0)
+, columnOffset(0)
+{}
+
+void Block::draw(int offsetX, int offsetY) const
 {
-    cellSize = 30;
-    rotationState = 0;
-    colors = getCellColors();
-    rowOffset = 0;
-    columnOffset = 0;
-}
-
-void Block::draw(int offsetX, int offsetY)
-{
-    std::vector<Position> tiles = getCellPositions();
-    for (Position item : tiles)
+    const auto& colors(getCellColors());
+    
+    const std::vector<Position> tiles = getCellPositions();
+    for (const Position& item : tiles)
```

Comment:
Use CONST REFERENCES if you just visit items for READING! A C++ Reference is implicitly realised with a Pointer -> No **expensive** object Copy is needed!